test/Classes/Foo.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace LaminasTest\Di\Classes;
+
+class Foo
+{
+    public function __construct(int|float $number)
```

Comment:
Let's call this `UnionTypeConstructorDependency`

Also: does this library even do DI with scalar types? I feel like these should be `object` types, so `stdClass|Countable` would work better, as an example.